corehq/apps/locations/adjacencylist.py

```diff
@@ -65,37 +87,91 @@ def cte_get_descendants(self, node, include_self=False):
 
         :param node: A model instance or a QuerySet or Q object querying
         the adjacency list model. If a QuerySet, it should query a
-        single value with something like `.values('pk')`. If Q the
+        single value with something like `.values('id')`. If Q the
         `include_self` argument will be ignored.
         :returns: A `QuerySet` instance.
         """
         parent_col = self.model.parent_id_attr
+        ordering_col = self.model.ordering_col_attr
 
+        discard_dups = False
         if isinstance(node, Q):
             where = node
+            discard_dups = True
         elif include_self:
             if isinstance(node, QuerySet):
-                where = Q(pk__in=node)
+                where = Q(id__in=node)
+                discard_dups = True
             else:
-                where = Q(pk=node.id)
+                where = Q(id=node.id)
         elif isinstance(node, QuerySet):
             where = Q(**{parent_col + "__in": node})
+            discard_dups = True
         else:
             where = Q(**{parent_col: node.id})
 
         def make_cte_query(cte):
             return self.filter(where).values(
-                "pk",
+                "id",
                 parent_col,
+                _cte_ordering=StrArray(ordering_col),
             ).union(
-                cte.join(self.model, **{parent_col: cte.col.pk}).values(
-                    "pk",
+                cte.join(self.model, **{parent_col: cte.col.id}).annotate(
+                    _cte_ordering=array_append(
+                        cte.col._cte_ordering,
+                        F(ordering_col),
+                    )
+                ).values(
+                    "id",
                     parent_col,
+                    "_cte_ordering",
                 ),
+                all=True,
             )
-
         cte = With.recursive(make_cte_query)
-        return cte.join(self.all(), pk=cte.col.pk).with_cte(cte)
+        ctes = [cte]
+
+        if discard_dups:
+            # Remove duplicates when the supplied Queryset or Q object
+            # may contain/match both parents and children. For a given
+            # id, retain the row with the longest path. TODO remove this
```

Comment:
it looks like this TODO never got addressed in this pr. does it need to be?

Reply:
Yes, I'm not planning to address this now since it's going to be a big task that may touch code in many places.